From 88cf0a6a819e7c4a27ecf78604eaf29cc4ed902c Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Fri, 7 Aug 2020 02:58:38 +0000
Subject: [PATCH] m25p80: fix m25p_probe() fail that caused while revert the
 commit 434ce69a7d117f91c2d1cd1ec9bb5eacdb70b7a3

pc : spi_nor_init_params+0x130/0xfa0
lr : spi_nor_scan+0x130/0xb50
sp : ffffff80115d33e0
x29: ffffff80115d33e0 x28: ffffff8011146000
x27: ffffff80115d3680 x26: ffffffc87882d888
x25: 0000000010d62088 x24: ffffffc87882d000
x23: ffffff80115d3794 x22: ffffffc87f7eef60
x21: ffffffc87a162700 x20: ffffffc87a162700
x19: ffffffc87882d888 x18: 0000000000000000
x17: 0000000000000000 x16: 0000000000000000
x15: 0000000000000000 x14: ffffffc87a9ca700
x13: 0000000000000000 x12: ffffff80112a9000
x11: ffffff8011166000 x10: ffffff80112a9450
x9 : 0000000000000000 x8 : 0000000000000400
x7 : 0000000000000003 x6 : ffffff80112a99de
x5 : 0000000000000000 x4 : 0000000000010101
x3 : 0000000000006b08 x2 : 0000000000000000
x1 : ffffff8010d62088 x0 : 0000000000000083
Call trace:
spi_nor_init_params+0x130/0xfa0
spi_nor_scan+0x130/0xb50
m25p_probe+0x1a0/0x218
spi_mem_probe+0x74/0xa8

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 drivers/mtd/devices/m25p80.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/mtd/devices/m25p80.c b/drivers/mtd/devices/m25p80.c
index c50888670250..dceeec32a4ca 100644
--- a/drivers/mtd/devices/m25p80.c
+++ b/drivers/mtd/devices/m25p80.c
@@ -190,6 +190,7 @@ static int m25p_probe(struct spi_mem *spimem)
 
 	spi_mem_set_drvdata(spimem, flash);
 	flash->spimem = spimem;
+	nor->spi = spi;
 
 	if (spi->mode & SPI_RX_OCTAL) {
 		hwcaps.mask |= SNOR_HWCAPS_READ_1_1_8;
-- 
2.17.1

